example_client/grpc_stateful_client.py

```diff
@@ -26,160 +26,356 @@
+        name_index += 1
+
+    # Consolidate reference scores
+    name_index = 0
+    reference_scores = dict()
+    for ark_score in reference_files:
+        output_name = output_names[name_index]
+        score_ojects = {k: m for k, m in ark_score}
+        reference_scores[output_name] = score_ojects
+        name_index += 1
+
+    # First ark file is the one we will iterate through for all input ark
+    # files per inference request input
+    data_iterator = dict()
+    for key, obj in input_files[0]:
+        data_iterator[key] = obj
+        # Validate reference output scores data for existing keys and shapes
+        for name in reference_scores:
+            score_objects = reference_scores[name]
+            if score_objects[key].shape[0] != obj.shape[0]:
+                print(
+                    "Error reference scores ark file doest not contain proper data for key {0} and shape {1}".format(
+                        key, obj.shape))
+                exit(1)
+
+    return data_iterator, input_names, output_names, input_data, reference_scores
+
+
+def validate_output(result, output_names):
+    # Validate model output
+    for output_name in output_names:
+        if output_name not in result.outputs:
+            print("ERROR: Invalid output name", output_name)
             print("Available outputs:")
-            for Y in result.outputs:
-                print(Y)
+            for o in result.outputs:
+                print(o)
+            return False
+    if 'sequence_id' not in result.outputs:
+        print("ERROR: Missing sequence_id in model output")
+        print("Available outputs:")
+        for o in result.outputs:
+            print(o)
+        return False
+    return True
+
+
+def main():
+    args = parse_arguments()
+    global debug_mode
+    debug_mode = int(args.get('debug'))
+
+    channel = grpc.insecure_channel(
+        "{}:{}".format(
+            args['grpc_address'],
+            args['grpc_port']))
+    stub = prediction_service_pb2_grpc.PredictionServiceStub(channel)
+    processing_times = np.zeros((0), int)
+    cw_l = int(args.get('cw_l'))
+    cw_r = int(args.get('cw_r'))
+    print('Context window left width cw_l: {}'.format(cw_l))
+    print('Context window right width cw_r: {}'.format(cw_r))
+    sequence_id = int(args.get('sequence_id'))
+    print('Starting sequence_id: {}'.format(sequence_id))
+    print('Start processing:')
+    print('Model name: {}'.format(args.get('model_name')))
+
+    data_iterator, input_names, output_names, input_data, reference_scores = prepare_processing_data(
+        args)
+
+    # Input control tokens
+    SEQUENCE_START = 1
+    SEQUENCE_END = 2
+
+    global_avg_rms_error_sum = 0.0
+
+    # Main inference loop
+    for key in data_iterator:
+        obj = data_iterator[key]
+        sequence_size = obj.shape[0]
+        print('\n\tInput name: {}'.format(key))
+        print_debug('\tInput in shape: {}'.format(obj.shape))
```

Comment:
Can we do this dim expansion here? 
So we get for example "Input in shape: (1, 490)" instead of "Input in shape: (39,490)" ? 
First dimension is sequence size and we already print it explicitly below.